src/System.IO.Pipelines/PipelineConnectionExtensions.cs

```diff
@@ -33,8 +31,16 @@ private static Task FlushAsync(WritableBuffer writeBuffer)
             var awaitable = writeBuffer.FlushAsync();
             if (awaitable.IsCompleted)
             {
-                awaitable.GetResult();
-                return _completedTask;
+                try
+                {
+                    awaitable.GetResult();
+                }
+                catch(Exception e)
```

Comment:
Why call `awaitable.GetResult()` to rethrow and then catch and wrap?

Why not test if `awaitable.IsFaulted` and return the exception on the task?